reactive/zeppelin.py

```diff
@@ -1,38 +1,53 @@
 # pylint: disable=unused-argument
+import hashlib
+
+from charmhelpers.core import unitdata
 from charms.reactive import when, when_not
 from charms.reactive import set_state, remove_state
 from charmhelpers.core import hookenv
-from charms.layer.apache_zeppelin import Zeppelin
+from charms.layer.apache_zeppelin import Zeppelin, ZeppelinAPI
 
 
 @when('spark.ready')
 @when_not('zeppelin.installed')
 def install_zeppelin(hadoop):
-    zepp = Zeppelin()
-    if zepp.verify_resources():
-        hookenv.status_set('maintenance', 'Installing Zeppelin')
-        zepp.install()
+    zepp = Zeppelin.get()
+    hookenv.status_set('maintenance', 'Installing Zeppelin')
+    if zepp.install():
+        zepp.setup_zeppelin()
         set_state('zeppelin.installed')
+    else:
+        hookenv.status_set('blocked', 'unable to fetch zeppelin resource')
 
 
 @when('zeppelin.installed', 'spark.ready')
 @when_not('zeppelin.started')
 def configure_zeppelin(spark):
     hookenv.status_set('maintenance', 'Setting up Zeppelin')
-    zepp = Zeppelin()
-    zepp.setup_zeppelin()
+    zepp = Zeppelin.get()
     zepp.configure_zeppelin()
     zepp.start()
     zepp.open_ports()
     set_state('zeppelin.started')
     hookenv.status_set('active', 'Ready')
 
 
+@when('zeppelin.started', 'spark.master.changed')
+def update_spark_master(spark):
+    zepp = Zeppelin.get()
+    api = ZeppelinAPI()
+    api.modify_interpreter('spark', properties={
```

Comment:
Need to make sure zepp is actually running before calling this, else the `requests.get` to localhost:9090 will fail.